wafamole/payloadfuzzer/sqlfuzzer.py

```diff
@@ -49,7 +49,8 @@ def logical_invariant(payload):
     :param payload:
     """
 
-    pos = re.search("(#|-- )", payload)
+    # pos = re.search("(#|-- )", payload)
+    pos = re.search(r"\b\w+(\s*(=|!=|<>|>|<|>=|<=)\s*|\s+(?i:LIKE|like|NOT LIKE|not like)\s+)\w+\b", payload)
```

Comment:
We could also change the `(?i:LIKE|like)` to simply `(?i:like)` since it's case insensitive (unless we want to do something very specific with this)

Also, are you sure the `(?i:LIKE)` construct works on all (recent) versions of Python? [Regex101](https://regex101.com/r/gMT9Yu/1) doesn't seem to support it (but I might be wrong here)

Reply:
Hello,
thanks for the review.
As for the first suggestion, yep you are right. I will add another commit to fix it.

As for the second one, I just re-tested it using Python 3.10.12 and it works for me.
According to the documentation, the pattern `(?i:...)` has been introduced in Python 3.6.
So, it matches the requirements of WAF-a-MoLE (Python > 3.7)